apollo-federation/src/schema/field_set.rs

```diff
@@ -92,21 +93,204 @@ pub(crate) fn parse_field_set(
+/// type from string to enum value when relevant (if the field set is invalid in some other way,
+/// this function skips the invalidity, assuming that later field set validation will catch this).
+/// In a future major release, we may error on such string/enum value mistakes instead.
+fn fix_string_enum_values_in_field_set(schema: &Valid<Schema>, field_set: &mut FieldSet) -> bool {
+    fix_string_enum_values_in_selection_set(schema, &mut field_set.selection_set)
+}
+
+fn fix_string_enum_values_in_selection_set(
+    schema: &Valid<Schema>,
+    selection_set: &mut executable::SelectionSet,
+) -> bool {
+    let mut is_modified = false;
+    for selection in selection_set.selections.iter_mut() {
+        match selection {
+            executable::Selection::Field(field) => {
+                let field = field.make_mut();
+                for argument in field.arguments.iter_mut() {
+                    let Some(type_) = field
+                        .definition
+                        .argument_by_name(&argument.name)
+                        .map(|def| &def.ty)
+                    else {
+                        continue;
+                    };
+                    let argument = argument.make_mut();
+                    if fix_string_enum_values_in_input_value(schema, type_, &mut argument.value) {
+                        is_modified = true;
+                    }
+                }
+                if fix_string_enum_values_in_directives(schema, &mut field.directives) {
+                    is_modified = true;
+                }
+                if fix_string_enum_values_in_selection_set(schema, &mut field.selection_set) {
+                    is_modified = true;
+                }
+            }
+            executable::Selection::FragmentSpread(fragment_spread) => {
+                let fragment_spread = fragment_spread.make_mut();
+                if fix_string_enum_values_in_directives(schema, &mut fragment_spread.directives) {
+                    is_modified = true;
+                }
+            }
+            executable::Selection::InlineFragment(inline_fragment) => {
+                let inline_fragment = inline_fragment.make_mut();
+                if fix_string_enum_values_in_directives(schema, &mut inline_fragment.directives) {
+                    is_modified = true;
+                }
+                if fix_string_enum_values_in_selection_set(
+                    schema,
+                    &mut inline_fragment.selection_set,
+                ) {
+                    is_modified = true;
+                }
+            }
+        }
+    }
+    is_modified
+}
+
+fn fix_string_enum_values_in_directives(
+    schema: &Valid<Schema>,
+    directives: &mut executable::DirectiveList,
+) -> bool {
+    let mut is_modified = false;
+    for directive in directives.iter_mut() {
+        let Some(directive_definition) = schema.directive_definitions.get(&directive.name) else {
+            return is_modified;
```

Comment:
Shouldn't we `continue` here?

Reply:
Yep we should use `continue` (it better aligns with how the rest of the code treats GraphQL invalidities), will fix.